core/keyserver/keyserver.go

```diff
@@ -189,13 +189,13 @@ func (s *Server) batchGetUserByRevision(ctx context.Context, sth *tpb.SignedLogR
 		Index:    indexes,
 		Revision: mapRevision,
 	})
-	if err != nil {
+	if st := status.Convert(err); st.Code() != codes.OK {
 		glog.Errorf("GetLeavesByRevision(%v, rev: %v): %v", d.Map.TreeId, mapRevision, err)
-		return nil, status.Errorf(codes.Internal, "Failed fetching map leaf")
+		return nil, status.Errorf(st.Code(), "Failed fetching map leaf")
```

Comment:
Depending on whether you check err != nil above, you could write a helper which takes an error and returns that error annotated with a string, passing along the existing status code if it was set.

Reply:
Indeed.  But that feels a little like it should really be part of `status`.
A true solution for this kind of thing will be coming in Go 2.0